overlord/install/install.go

```diff
@@ -254,8 +255,25 @@ func PrepareEncryptedSystemData(model *asserts.Model, installKeyForRole map[stri
 	// make note of the encryption keys
 	trustedInstallObserver.SetBootstrappedContainers(dataBootstrappedContainer, saveBootstrappedContainer)
 
-	if err := saveKeys(model, installKeyForRole); err != nil {
-		return err
+	// XXX: is the asset cache problematic from initramfs?
+	// keep track of recovery assets
+	if err := trustedInstallObserver.ObserveExistingTrustedRecoveryAssets(boot.InitramfsUbuntuSeedDir); err != nil {
+		return fmt.Errorf("cannot observe existing trusted recovery assets: %v", err)
+	}
```

Comment:
bit confused by this addition, is this because of the initramfs case? this is called in handlers_install.go already before calling PrepareEncryptedSystemData  ? something more needs to be clarified here and maybe called removed there?

Reply:
Botched rebase. It was a conflict with 2034c7edb2001174986eeb2e14ed0fe0b50e91d2